llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -163,6 +190,16 @@ void Function::dumpNameAndArgs(raw_ostream &OS) const {
 void Function::dump(raw_ostream &OS) const {
   dumpNameAndArgs(OS);
   OS << " {\n";
+  auto *LLVMF = cast<llvm::Function>(Val);
+  for (llvm::BasicBlock &LLVMBB : *LLVMF) {
+    auto *BB = cast_or_null<BasicBlock>(Ctx.getValue(&LLVMBB));
+    if (BB == nullptr)
+      OS << "NULL";
+    else
+      OS << *BB;
+    if (std::next(LLVMBB.getIterator()) != LLVMF->end())
```

Comment:
the `interleave` functions in `STLExtras.h` might be useful here (or maybe this is fine)